python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1407,3 +1477,496 @@ def _compute_model_params(self, wcsf):
+
+        outputWCSs = self._make_outputs(wcsf, inputVisitSummaries, exposureInfo)
+        outputCatalog = wcsf.getOutputCatalog()
+        starCatalog = wcsf.getStarCatalog()
+
+        return pipeBase.Struct(
+            outputWCSs=outputWCSs, fitModel=wcsf, outputCatalog=outputCatalog, starCatalog=starCatalog
+        )
+
+    def _prep_sky(self, inputVisitSummaries):
+        """Get center and radius of the input tract. This assumes that all
+        visits will be put into the same `wcsfit.Field` and fit together.
+
+        Paramaters
+        ----------
+        inputVisitSummaries : `list` of `lsst.afw.table.ExposureCatalog`
+            List of catalogs with per-detector summary information.
+
+        Returns
+        -------
+        fields : `wcsfit.Fields`
+            Object with field information.
+        fieldRegions : `dict` [`int`, `lsst.sphgeom.ConvexPolygon`]
+            Dictionary of regions encompassing each group of input visits,
+            keyed by an arbitrary index.
+        """
+        allDetectorCorners = []
+        mjds = []
+        radecs = []
+        radii = []
+        for visSum in inputVisitSummaries:
+            detectorCorners = [
+                lsst.geom.SpherePoint(ra, dec, lsst.geom.degrees).getVector()
+                for (ra, dec) in zip(visSum["raCorners"].ravel(), visSum["decCorners"].ravel())
+            ]
+            allDetectorCorners.append(detectorCorners)
+
+            # Get center and approximate radius of field of view
+            hull = lsst.sphgeom.ConvexPolygon.convexHull(detectorCorners)
+            boundingCircle = lsst.sphgeom.ConvexPolygon.convexHull(detectorCorners).getBoundingCircle()
+            center = lsst.geom.SpherePoint(boundingCircle.getCenter())
+            ra = center.getRa().asDegrees()
+            dec = center.getDec().asDegrees()
+            radecs.append([ra, dec])
+            radius = boundingCircle.getOpeningAngle()
+            radii.append(radius)
+
+            obsDate = visSum[0].getVisitInfo().getDate()
+            obsMJD = obsDate.get(obsDate.MJD)
+            mjds.append(obsMJD)
+
+        # Find groups of visits where any one of the visits overlaps another by
+        # a given fraction of the field-of-view radius.
+        distance = self.config.visitOverlap * np.median(radii)
+        clustering = AgglomerativeClustering(
+            distance_threshold=distance.asDegrees(), n_clusters=None, linkage="single"
+        )
+        clusters = clustering.fit(np.array(radecs))
+
+        medianMJD = np.median(mjds)
+        medianEpoch = astropy.time.Time(medianMJD, format="mjd").decimalyear
+
+        fieldNames = []
+        fieldRAs = []
+        fieldDecs = []
+        epochs = []
+        fieldRegions = {}
+
+        for i in range(clusters.n_clusters_):
+            fieldInd = clusters.labels_ == i
+            fieldDetectors = sum([allDetectorCorners[f] for f, fInd in enumerate(fieldInd) if fInd], [])
```

Comment:
Yeah, I think this syntax was new to me and came from stackOverflow. I added "# Concatenate the lists of all detector corners that are in this field". I think `np.concatenate` would do almost the same thing, but I didn't want `fieldDetectors` to be an `ndarray`.

Reply:
The `sum()` operation turns a list of lists (a list of corners for a list of detectors) into one long list because this is what `lsst.sphgeom.ConvexPolygon.convexHull()` requires.